server/meshmodel/helper.go

```diff
@@ -0,0 +1,144 @@
+		doneSignal:       make(chan bool),
+		errorChan:        make(chan error),
+	}
+}
+
+// seed the local meshmodel components
+func (ch *ComponentHelper) SeedComponents() {
+	// Watch channels and register components and relationships with the registry manager
+	go ch.watchComponents()
+
+	// Read component and relationship definitions from files and send them to respective channels
+	ch.generateComponents("../meshmodel/components")
+	ch.generateRelationships("../meshmodel/components")
+	ch.doneSignal <- true
+}
+
+func (ch *ComponentHelper) generateComponents(pathToComponents string) {
+	path, err := filepath.Abs(pathToComponents)
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while getting absolute path for generating components")
+		return
+	}
+
+	err = filepath.Walk(path, func(path string, info fs.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+
+		if !info.IsDir() {
+			var comp v1alpha1.ComponentDefinition
+			byt, err := os.ReadFile(path)
+			if err != nil {
+				ch.errorChan <- errors.Wrapf(err, fmt.Sprintf("unable to read file at %s", path))
+				return nil
+			}
+			err = json.Unmarshal(byt, &comp)
+			if err != nil {
+				ch.errorChan <- errors.Wrapf(err, fmt.Sprintf("unmarshal json failed for %s", path))
+				return nil
+			}
+			if comp.Metadata != nil && comp.Metadata["published"] == true {
+				utils.WriteSVGsOnFileSystem(&comp)
+				ch.componentChan <- comp
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while generating components")
+	}
+	return
+}
+
+func (ch *ComponentHelper) generateRelationships(pathToComponents string) {
+	path, err := filepath.Abs(pathToComponents)
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while getting absolute path for generating relationships")
+		return
+	}
+
+	err = filepath.Walk(path, func(path string, info fs.FileInfo, err error) error {
+		if info == nil {
+			return nil
+		}
+		if !info.IsDir() {
+			var rel v1alpha1.RelationshipDefinition
+			byt, err := os.ReadFile(path)
+			if err != nil {
+				ch.errorChan <- errors.Wrapf(err, fmt.Sprintf("unable to read file at %s", path))
+				return nil
+			}
+			err = json.Unmarshal(byt, &rel)
+			if err != nil {
+				ch.errorChan <- errors.Wrapf(err, fmt.Sprintf("unmarshal json failed for %s", path))
+				return nil
+			}
+			ch.relationshipChan <- rel
+		}
+		return nil
+	})
+	if err != nil {
+		ch.errorChan <- errors.Wrapf(err, "error while generating relationships")
+	}
+	return
+}
+
+func (ch *ComponentHelper) watchComponents() {
+	var err error
+	for {
+		select {
+		case comp := <-ch.componentChan:
+			err = ch.regManager.RegisterEntity(meshmodel.Host{
+				Hostname: ArtifactHubComponentsHandler,
+			}, comp)
+		case rel := <-ch.relationshipChan:
+			err = ch.regManager.RegisterEntity(meshmodel.Host{
+				Hostname: ArtifactHubComponentsHandler,
+			}, rel)
+		case mhErr := <-ch.errorChan:
+			fmt.Println("err: ", mhErr.Error())
```

Comment:
@Azanul Do not use println. Create appropriate meshkit error. And pass the meshkit logger initialized in main.go to ComponentHelper struct.